core/trino-main/src/main/java/io/trino/operator/SimpleTableExecuteOperator.java

```diff
@@ -0,0 +1,152 @@
+                int operatorId,
+                PlanNodeId planNodeId,
+                Metadata metadata,
+                Session session,
+                TableExecuteHandle executeHandle)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.metadata = requireNonNull(metadata, "planNodeId is null");
+            this.session = requireNonNull(session, "planNodeId is null");
+            this.executeHandle = requireNonNull(executeHandle, "executeHandle is null");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+            OperatorContext context = driverContext.addOperatorContext(operatorId, planNodeId, SimpleTableExecuteOperator.class.getSimpleName());
+            return new SimpleTableExecuteOperator(
+                    context,
+                    metadata,
+                    session,
+                    executeHandle);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            return new SimpleTableExecuteOperatorOperatorFactory(
+                    operatorId,
+                    planNodeId,
+                    metadata,
+                    session,
+                    executeHandle);
+        }
+    }
+
+    private final OperatorContext operatorContext;
+    private final Metadata metadata;
+    private final Session session;
+    private final TableExecuteHandle executeHandle;
+
+    private boolean finished;
+
+    public SimpleTableExecuteOperator(
+            OperatorContext operatorContext,
+            Metadata metadata,
+            Session session,
+            TableExecuteHandle executeHandle)
+    {
+        this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.session = requireNonNull(session, "session is null");
+        this.executeHandle = requireNonNull(executeHandle, "executeHandle is null");
+    }
+
+    @Override
+    public OperatorContext getOperatorContext()
+    {
+        return operatorContext;
+    }
+
+    @Override
+    public boolean needsInput()
+    {
+        return false;
+    }
+
+    @Override
+    public void addInput(Page page)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Page getOutput()
+    {
+        if (finished) {
+            return null;
+        }
+
+        metadata.executeTableExecute(session, executeHandle);
+        PageBuilder page = new PageBuilder(1, TYPES);
+        BlockBuilder rowsBuilder = page.getBlockBuilder(0);
+        page.declarePosition();
+        BIGINT.writeLong(rowsBuilder, 0);
```

Comment:
do not emit any rows. Also you can declare empty page as a constant, not construct it on each call to getOutput.